drivers/serial/Kconfig.nrfx_uart_instance

```diff
@@ -68,7 +68,7 @@ config UART_$(nrfx_uart_num)_NRF_ASYNC_LOW_POWER
 	depends on HAS_HW_NRF_UARTE$(nrfx_uart_num)
 	depends on UART_ASYNC_API
 	depends on UART_NRFX_UARTE_LEGACY_SHIM
-	default y
+	default y if !PM_DEVICE
```

Comment:
This should be rather `PM_DEVICE_RUNTIME`. But maybe, instead of this, it would be better to mention in help that `PM_DEVICE_RUNTIME` overrides this setting, I mean that the instance will be appropriately disabled only if the runtime PM is enabled for it.

Reply:
I've added note in help and do not set low power flag when runtime PM is used.

Reply:
After more testing i came back to a conclusion that low power cannot be mixed with PM device. So I restored `if !PM_DEVICE` and note in  help. The reason is that when explicit device PM is used then we have two places where UART can be disabled (automatically in the interrupt and in the suspend action) and it is hard to manage. Given that we are going towards runtime device PM it does not make much sense to spend effort of feature that won't be used.

Reply:
Agreed. It's not worth spending more more time on this now. Excluding the old low-power mechanism entirely when PM is used is actually a "safe path" so let's take such approach.